src/codeCache.cpp

```diff
@@ -162,18 +162,20 @@ const char* CodeCache::binarySearch(const void* address, LinearAllocator* alloca
     if (low > 0 && (_blobs[low - 1]._start == _blobs[low - 1]._end || _blobs[low - 1]._end == address)) {
         return _blobs[low - 1]._name;
     }
-
+    char buf[512];
+    // add 1 for the null terminator
+    int n;
+    char *allocated;
     if (_add_build_ids && _build_id) {
-        char buf[512];
-        // add 1 for the null terminator
-        const int n = 1 + snprintf(buf, sizeof(buf), "%s %s+%p_[bid]", _name, _build_id, (void*)((char*)address - (char*)_min_address));
-        char *allocated;
-        // if snprintf didn't overflow, and we manage to allocate a suitable buffer (from the LinearAllocator, not malloc,
-        // since this is signal context and malloc can deadlock) - then use them.
-        if (n < sizeof(buf) && (allocated = (char*)allocator->alloc(n)) != NULL) {
-            memcpy(allocated, buf, n);
-            return allocated;
-        }
+        n = 1 + snprintf(buf, sizeof(buf), "(%s) %s+%p_[bid]", _name, _build_id, (void*)((char*)address - (char*)_min_address));
+    } else {
+        n = 1 + snprintf(buf, sizeof(buf), "(%s)", _name);
+    }
+    // if snprintf didn't overflow, and we manage to allocate a suitable buffer (from the LinearAllocator, not malloc,
+    // since this is signal context and malloc can deadlock) - then use them.
+    if (n < sizeof(buf) && (allocated = (char*)allocator->alloc(n)) != NULL) {
+        memcpy(allocated, buf, n);
+        return allocated;
```

Comment:
Here we allocate via the `LinearAllocator` and this value eventually returns to `convertNativeTrace`. From there it's passed to `NativeFunc::isMarked` which expects a `NativeFunc` struct to precede the string itself. So this code doesn't seem legit anymore (and as I said we don't use the buildids feature now; it does have a test but the test may still pass despite accessing memory incorrectly). I wrote the buildids feature before `NativeFunc` was added, so it probably "got broken" along one of the rebases.

I think - 
1. I will revert the buildids for now, because it is clearly broken if enabled :/ And disable the test in gProfiler
2. You can meanwhile change your branch to introduce the `( )` in another site and not in a site where `NativeFunc` is expected. If we can't find any, we can revert to the previous way. Also, another idea that I have - instead of changing async-profiler, we can change `java.py` in gProfiler to make the modifications. It will just use `lib` in the annotations, and change the `lib`name` to `name (lib)` itself, after we call `parse_one_collapsed`. Maybe that was the best idea to begin with... we should have thought about it earlier lol

Reply:
I removed the feature in gProfiler - https://github.com/Granulate/gprofiler/pull/558 you can rebase on the new async-profiler version so you won't have the buildids code to handle in your branch.

Reply:
I agree and did the rebase.